xds/internal/balancer/lrs/balancer.go

```diff
@@ -199,31 +200,40 @@ type xdsClientWrapper struct {
 	clusterName      string
 	edsServiceName   string
 	lrsServerName    string
-	load             *loadStoreWrapper
+	// loadOriginal is the load.Store for reporting loads to lrsServerName. It's
+	// returned by the client.
+	loadOriginal *load.Store
+	// loadWrapper is a wrapper with loadOriginal, with clusterName and
+	// edsServiceName. It's used children to report loads.
+	loadWrapper *loadStoreWrapper
 }
 
 func newXDSClientWrapper() *xdsClientWrapper {
 	return &xdsClientWrapper{
-		load: &loadStoreWrapper{},
+		loadWrapper: &loadStoreWrapper{},
 	}
 }
 
 // update checks the config and xdsclient, and decides whether it needs to
 // restart the load reporting stream.
-func (w *xdsClientWrapper) update(newConfig *lbConfig, attr *attributes.Attributes) {
+func (w *xdsClientWrapper) update(newConfig *lbConfig, attr *attributes.Attributes) error {
 	var (
 		restartLoadReport bool
 		updateLoadStore   bool
 	)
-	if attr != nil {
-		if clientFromAttr, _ := attr.Value(xdsinternal.XDSClientID).(xdsClientInterface); clientFromAttr != nil {
-			if w.c != clientFromAttr {
-				// xds client is different, restart.
-				restartLoadReport = true
-				updateLoadStore = true
-				w.c = clientFromAttr
-			}
-		}
+
+	if attr == nil {
+		return fmt.Errorf("failed to get xdsClient from attributes: attributes is nil")
```

Comment:
Should these errors be prefixed with `xds:` or `lrs:`. I'm a little confused about the policy for these error message. I guess we dont have to do that for log statements since the prefix logger takes care of it.

Reply:
Added `lrs:`.

The caller (which is the implementation of the balancer interface) could wrap. But that will be only to add the prefix, and seems to be not worth it. I added it here.